mesh/pmesh.cpp

```diff
@@ -3190,6 +3193,53 @@ int ParMesh::GetNFbyType(FaceType type) const
    return Mesh::GetNFbyType(type);
 }
 
+void ParMesh::GenerateBoundaryElements()
+{
+   for (auto &b : boundary)
+   {
+      FreeElement(b);
+   }
+
+   if (Dim == 3)
+   {
+      delete bel_to_edge;
+      bel_to_edge = NULL;
+   }
+
+   const Array<int> *s2l_face;
+   if (Dim == 1) { s2l_face = &svert_lvert; }
+   else if (Dim == 2) { s2l_face = &sedge_ledge; }
+   else { s2l_face = &sface_lface; }
+
+   // mark the shared local elements
+   Array<bool> l2s_marker(GetNumFaces());
+   l2s_marker = false;
+   for (int l : *s2l_face) { l2s_marker[l] = true; }
+
+   // count the 'NumOfBdrElements'
+   NumOfBdrElements = 0;
+   for (int i = 0; i < faces_info.Size(); i++)
+   {
+      const auto &fi = faces_info[i];
+      if (fi.Elem2No < 0 && !l2s_marker[i])
```

Comment:
I think this can be simplified to check `fi.Elem2Inf`. Or just use `FaceIsTrueInterior` in the `Mesh::GenerateBoundaryElements` implementation instead of just checking `fi.Elem2No` (which is like checking `FaceIsInterior`).

Reply:
No, it does not work because this encoded information about faces is available only after `ExchangeFaceNbrData()` has been called (as the docstring of `FaceIsInterior` mentions). The faces at this point just come from `Mesh::GenerateFaces()`, so there are not many options really.